lib/note-content-editor.jsx

```diff
@@ -89,6 +88,56 @@ function outdentCurrentBlock( editorState ) {
 	);
 }
 
+function finishList( editorState ) {
+	// remove `- ` from the current line
+	const withoutBullet = EditorState.push(
+		editorState,
+		Modifier.removeRange(
+			editorState.getCurrentContent(),
+			editorState.getSelection().merge( {
+				anchorOffset: 0,
+				focusOffset: getCurrentBlock( editorState ).getLength(),
+			} )
+		),
+		'remove-range'
+	);
+
+	// move selection to the start of the line
+	return EditorState.forceSelection(
+		withoutBullet,
+		withoutBullet.getCurrentContent().getSelectionAfter()
+	);
+}
+
+function continueList( editorState, listItemMatch ) {
```

Comment:
I'm suspicious of the implicit state being passed in `listItemMatch` and curious why we need it. it requires a RegExp match object but only uses the first item. what if we required this to be the bullet+prefix and called it something like that

```js
function continueList( editorState, listPrefix ) { … }
```

then the calling code would read like this:
```js
const listItem = matcher.match( line );
if ( listItem ) {
	return … this.continueList( editorState, listItem[ 0 ] );
}
```

or better yet use some destructuring and comments…
```js
// list item has prefix up to the bullet and following space then arbitrary text
//  - thing
const listItem = pattern.match( line );
if ( listItem ) {
	const [ listPrefix ] = listItem;
	return this.handle… this.continueList( editorState, listPrefix );
}
```